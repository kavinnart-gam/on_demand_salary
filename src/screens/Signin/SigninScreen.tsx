import React, {useState, useContext} from 'react';
import {
  StyleSheet,
  View,
  Text,
  TextInput,
  TouchableOpacity,
  ActivityIndicator,
} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import {asyncStorage} from '../../utils';
import common from '../../utils/common';
import {AuthContext} from '../../navigator/AppNavigator';
import {setToken} from '../../slices/authSlice';
import {useDispatch} from 'react-redux';

function SigninScreen({navigation}: any) {
  const dispatch = useDispatch();
  const {login, isLoading} = useContext(AuthContext);
  const [phoneNumber, setphoneNumber] = useState<string>('');

  const onPressLogin = async () => {
    const response = await login(phoneNumber);

    if (response?.data?.token) {
      await asyncStorage.setDataToAsyncStorage({
        key: 'idToken',
        value: response?.data?.token,
      });
      dispatch(setToken(response?.data?.token));
      navigation.navigate('VerifyOtp');
    }
  };

  return (
    <SafeAreaView style={styles.container}>
      {isLoading ? (
        <ActivityIndicator size="large" color="#000" />
      ) : (
        <>
          <View style={styles.inputView}>
            <TextInput
              {...common.testID('INPUT_TXT_MOBILE_NUMBER')}
              style={styles.inputText}
              placeholder="Mobile Phone Number"
              placeholderTextColor="#003f5c"
              dataDetectorTypes="phoneNumber"
              keyboardType="phone-pad"
              textContentType="telephoneNumber"
              returnKeyType="done"
              maxLength={10}
              onChangeText={text => setphoneNumber(text)}
            />
          </View>
          <TouchableOpacity
            {...common.testID('BTN_SIGN_IN')}
            disabled={phoneNumber.length === 0}
            style={[
              styles.loginBtn,
              phoneNumber.length < 10 && styles.disableBtn,
            ]}
            onPress={onPressLogin}>
            <Text style={styles.loginText}>LOGIN </Text>
          </TouchableOpacity>
        </>
      )}
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#debbb0',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputView: {
    width: '80%',
    backgroundColor: 'white',
    borderRadius: 25,
    height: 50,
    marginBottom: 20,
    justifyContent: 'center',
    padding: 20,
  },
  inputText: {
    height: 50,
    color: '#000000',
  },
  disableBtn: {
    backgroundColor: '#E5EAEF',
    borderColor: '#6A6A6A',
    borderWidth: 1,
  },
  loginBtn: {
    width: '80%',
    backgroundColor: '#000000',
    color: '#FFFFFF',
    borderRadius: 25,
    height: 50,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 40,
    marginBottom: 10,
  },
  loginText: {
    color: '#FFFFFF',
  },
});

export default SigninScreen;
