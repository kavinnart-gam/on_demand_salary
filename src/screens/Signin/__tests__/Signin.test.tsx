import {render, fireEvent} from '@testing-library/react-native';
import SigninScreen from '../SigninScreen';
describe('Signin', () => {
  it('Sign screen is match snapshot', () => {
    const result = render(<SigninScreen />);
    expect(result).toMatchSnapshot();
  });

  it('Sign screen component should be render correctly', () => {
    const {getByTestId} = render(<SigninScreen />);
    const buttonSignin = getByTestId('BTN_SIGN_IN');
    const inputMobileNumber = getByTestId('INPUT_TXT_MOBILE_NUMBER');

    expect(buttonSignin).toBeTruthy();
    expect(inputMobileNumber).toBeTruthy();
  });

  it('When fill phone number signin button can be enable', () => {
    const {getByTestId} = render(<SigninScreen />);

    const buttonSignin = getByTestId('BTN_SIGN_IN');
    expect(buttonSignin.props.style.backgroundColor).toEqual('#E5EAEF');

    const inputMobileNumber = getByTestId('INPUT_TXT_MOBILE_NUMBER');
    fireEvent.changeText(inputMobileNumber, '0835654787');
    expect(buttonSignin.props.style.backgroundColor).toEqual('#000000');
  });
});
