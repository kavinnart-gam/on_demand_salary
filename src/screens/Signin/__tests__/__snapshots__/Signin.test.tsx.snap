// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin Sign screen is match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#debbb0",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      accessibilityLabel="INPUT_TXT_MOBILE_NUMBER"
      accessible={true}
      dataDetectorTypes="phoneNumber"
      keyboardType="phone-pad"
      maxLength={10}
      onChangeText={[Function]}
      placeholder="Mobile Phone Number"
      placeholderTextColor="#003f5c"
      returnKeyType="done"
      style={
        {
          "color": "#000000",
          "height": 50,
        }
      }
      testID="INPUT_TXT_MOBILE_NUMBER"
      textContentType="telephoneNumber"
    />
  </View>
  <View
    accessibilityLabel="BTN_SIGN_IN"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E5EAEF",
        "borderColor": "#6A6A6A",
        "borderRadius": 25,
        "borderWidth": 1,
        "color": "#FFFFFF",
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 40,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="BTN_SIGN_IN"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
        }
      }
    >
      LOGIN 
    </Text>
  </View>
</RNCSafeAreaView>
`;
